### PR TITLE
Add `NequIP` & `Allegro` to matbench leaderboard

### PR DESCRIPTION
**Pull request by kavanase**
## Description

This PR adds initial foundation potentials from the `NequIP` GNN and `Allegro` architectures, including both compliant and non-compliant (OMat) models. 

## Checklist

Please check the following items before submitting your PR:

- [x] I created a new folder and YAML metadata file `models/<arch_name>/<model_variant>.yml` for my submission. `arch_name` is the name of the architecture and `model_variant.yml` includes things like author details, training set names and important hyperparameters.
- [x] I added the new model as a new attribute [`Model.<arch_name>` enum](https://github.com/janosh/matbench-discovery/blob/57d0d0c8a14cd317/matbench_discovery/enums.py#L274) on the `Model` enum in `enums.py`.
- [x] I uploaded the energy/force/stress model prediction file for the WBM test set to Figshare or another cloud storage service (`<yyyy-mm-dd>-<model_variant>-preds.csv.gz`).
- [x] I uploaded the model-relaxed structures file to Figshare or another cloud storage service in [JSON lines format](https://jsonlines.org) (`<yyyy-mm-dd>-wbm-IS2RE-FIRE.jsonl.gz`). JSON Lines allows fast loading of small numbers of structures with `pandas.read_json(lines=True, nrows=100)` for inspection.
- [x] I uploaded the phonon predictions to Figshare or another cloud storage service (`<yyyy-mm-dd>-kappa-103-FIRE-<values-of-dist|fmax|symprec>.gz`).
- [x] I included the urls to the Figshare files in the YAML metadata file (`models/<arch_name>/<model_variant>.yml`). If not using Figshare I have included the urls to the cloud storage service in the description of the PR.
- [x] I included the test script (`test_<arch_name>_<task>.py` for `task` in `discovery`, `kappa`, `diatomics`) that generated the prediction files.

## Additional Information (Optional)

- [x] I included a training script (`train_<arch_name>.py`) if I trained a model specifically for this benchmark.
- [x] I included a `readme.md` with additional details about my model.

@coderabbitai ignore

### CODE REVIEW

**Comment by janosh** [comment]
very nice work! 👍  thanks a lot for this submission @kavanase! 🙏 i'm very excited to add NequiP and Allegro to the benchmark!

thanks for reporting the `data/wbm/compile_wbm_test_set.py` import error. should be fixed in [f3419a2](https://github.com/janosh/matbench-discovery/pull/282/commits/f3419a26385c0532660679b2f6dd06644608601e)

re the failing `scripts/evals/kappa_srme.py`, I haven't taken a closer look yet but at first glance seems like the column names are unexpected (e.g. `mp_id` instead of `material_id`). i'll take a closer look. no need for you to debug

**Comment by janosh** [comment]
this script seems to mostly bring the kappa pred files into the expected format (except there's still an extra nesting level that doesn't play well with `pd.read_json`). need to change trains now but will have another go later.

<details><summary>standardize_kappa_preds.py</summary>

```py
"""Convert kappa predictions to Matbench format."""

import json
import os
import shutil

import pandas as pd
from pymatviz.enums import Key

from matbench_discovery import today
from matbench_discovery.enums import MbdKey, Model

DRY_RUN = False
col_map = {
    "mp_id": Key.mat_id,
    "name": Key.formula,
    "max_stress": Key.max_stress,
    "reached_max_steps": "reached_max_steps",
    "imaginary_freqs": Key.has_imag_ph_modes,
    "frequencies": Key.ph_freqs,
    "kappa_TOT_RTA": MbdKey.kappa_tot_rta,
    "kappa_P_RTA": MbdKey.kappa_p_rta,
    "kappa_C": MbdKey.kappa_c,
    "weights": Key.mode_weights,
    "qpoints": Key.q_points,
    "mode_kappa_TOT": MbdKey.mode_kappa_tot_rta,
    "kappa_TOT_ave": MbdKey.kappa_tot_avg,
    "initial_space_group_number": Key.spg_num,
    "mode_kappa_TOT_ave": MbdKey.mode_kappa_tot_avg,
}
cols_to_drop = [
    "DFT_kappa_TOT_ave",
    "SRD",
    "SRE",
    "SRME",
    "errors",
    "error_traceback",
    "symprec_tests",
    "redirected_to_symm",
]

models_updated: dict[str, dict[str, str | list[str]]] = {}


for model in Model:
    if model.kappa_103_path is None or not os.path.isfile(model.kappa_103_path):
        print(f"{model.label=} kappa_103_path={model.kappa_103_path!r} does not exist")
        continue

    df_ml = pd.read_json(model.kappa_103_path)
    if MbdKey.mode_kappa_tot_avg in df_ml:
        print(f"{model.label=} already processed")
        continue  # skip if already processed

    old_cols = df_ml.columns.copy()
    df_ml = df_ml.rename(columns=col_map)

    renamed_cols = {
        str(old_col): str(new_col)
        for old_col, new_col in zip(old_cols, df_ml)
        if old_col != new_col
    }
    old_cols = df_ml.columns.copy()
    df_ml = df_ml.drop(columns=cols_to_drop, errors="ignore")
    dropped_cols = [col for col in old_cols if col not in df_ml]

    # if dataframe was modified, backup the original file
    if renamed_cols or dropped_cols:
        models_updated[model.key] = {
            "renamed_cols": renamed_cols,
            "dropped_cols": dropped_cols,
        }
        if not DRY_RUN:
            shutil.copy(model.kappa_103_path, f"{model.kappa_103_path}.bak")
            df_ml.to_json(model.kappa_103_path)
            print(f"{model.label=} new column names: {df_ml.columns}")
    else:
        print(f"{model.label=} no changes")


print(f"\n\n{len(models_updated)=}")

for model_key, actions in models_updated.items():
    print(f"{model_key=} {actions=}")

# write updated models to file
with open(f"{today}-kappa-model-files-updated.json", mode="w") as file:
    json.dump(models_updated, file, indent=2)

```
</details>

**Comment by janosh** [comment]
@kavanase very nearly ready to merge. there are some TODOs left

https://github.com/janosh/matbench-discovery/blob/9aea79466865f4e895e164152cc51840becc0c01/models/allegro/allegro-MP-L-0.1.yml#L56

can we address those now that the checkpoints are up https://www.nequip.net/models/mir-group/Allegro-MP-L:0.1?

**Comment by kavanase** [comment]
> @kavanase very nearly ready to merge. there are some TODOs left
> 
> https://github.com/janosh/matbench-discovery/blob/9aea79466865f4e895e164152cc51840becc0c01/models/allegro/allegro-MP-L-0.1.yml#L56
> 
> can we address those now that the checkpoints are up https://www.nequip.net/models/mir-group/Allegro-MP-L:0.1?

Fixed now! I missed this entry for the Allegro models on the first draft 🤦

**Comment by kavanase** [comment]
@janosh thanks very much for pushing through with this!
I see the [Tests / scripts (scripts/evals/discovery.py) (pull_request)](https://github.com/janosh/matbench-discovery/actions/runs/17544869648/job/49823918843?pr=282) appears as passing, but looking at the outputs it's showing download errors for the updated NequIP/Allegro figshare files?

image

**Review by janosh — APPROVED** [review]
thanks again for this submission @kavanase! outstanding work 👍 🚀 

been hoping for the day when Allegro and Nequip are added to the leaderboard for almost 2 years now! so many people kept asking me why those we're missing and all I could say is lack of time...
very glad that's now over 🙏 

> but looking at the outputs it's showing download errors for the updated NequIP/Allegro figshare files?

i think i just forgot to re-publish the Figshare article after uploading these files. should be good now
